From: ipsec-tools packagers <team+ipsec-tools@tracker.debian.org>
Date: Sun, 7 Jun 2020 15:41:18 +0200
Subject: include-stdint

===================================================================
---
 src/libipsec/libpfkey.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/libipsec/libpfkey.h b/src/libipsec/libpfkey.h
index a213aac..c614b3a 100644
--- a/src/libipsec/libpfkey.h
+++ b/src/libipsec/libpfkey.h
@@ -55,6 +55,9 @@ struct sadb_alg;
 
 /* Accomodate different prototypes in <netinet6/ipsec.h> */
 #include <sys/types.h>
+#ifdef HAVE_STDINT_H
+#include <stdint.h>
+#endif
 #include PATH_IPSEC_H
 
 #ifndef HAVE_IPSEC_POLICY_T
